### PR TITLE
Support HOSTING_ENV === 'aws' option

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Support HOSTING_ENV === 'aws' option

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/1165

### CODE REVIEW

File: common/network-util.js [hunk 1]
```diff
@@ -8,10 +8,10 @@ const extIp = require('ext-ip')();
...
 const DB = require('../db');
 const { JSON_RPC_METHODS } = require('../json_rpc/constants');
-const GCP_EXTERNAL_IP_URL = 'http://metadata.google.internal/computeMetadata/v1/instance' +
-    '/network-interfaces/0/access-configs/0/external-ip';
-const GCP_INTERNAL_IP_URL = 'http://metadata.google.internal/computeMetadata/v1/instance' +
-    '/network-interfaces/0/ip';
+const GCP_EXTERNAL_IP_URL = 'http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip';
+const GCP_INTERNAL_IP_URL = 'http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ip';
+const AWS_EXTERNAL_IP_URL = 'http://169.254.169.254/latest/meta-data/public-ipv4';
+const AWS_INTERNAL_IP_URL = 'http://169.254.169.254/latest/meta-data/local-ipv4';
```
Comment (minsulee2) [inline common/network-util.js:14]: Is it OK with IP exposure?

Comment (platfowner) [reply common/network-util.js:14]: It's not our ip address. It's a public ip available for all aws instances.
(ref: https://stackoverflow.com/questions/38679346/get-public-ip-address-on-current-ec2-instance)

File: common/network-util.js [hunk 2]
```diff
@@ -108,6 +108,17 @@ function getIpAddress(internal = false) {
         CommonUtil.finishWithStackTrace(
             logger, `Failed to get ip address: ${JSON.stringify(err, null, 2)}`);
       });
+    } else if (NodeConfigs.HOSTING_ENV === 'aws') {
```
Comment (ehgmsdk20) [inline common/network-util.js:111]: It seems that there are several HOSTING_ENVs: local, comcom, gcp.
Other files also refer to NodeConfigs.HOSTING_ENV, so how about storing these in the constant.js?

**Review by ehgmsdk20 — APPROVED** [review]
LGTM with a suggestion, thanks!

File: common/network-util.js (hunk 2 above)
Comment (ehgmsdk20) [reply common/network-util.js:111]: Also, I am wondering what does 'comcom' mean as HOSTING_ENV?

Comment (platfowner) [reply common/network-util.js]: Just added HostingEnvs enum definition. PTAL

'comcom' is ComCom On-Premise.

File: common/constants.js [hunk 3]
```diff
@@ -131,6 +131,18 @@ function setNodeConfigs() {
...
 
 // ** Enums **
+/**
+ * Hosting Environments.
+ *
+ * @enum {string}
+ */
+const HostingEnvs = {
+  AWS: 'aws',  // Amazon AWS
+  COMCOM: 'comcom',  // ComCom On-Premise
```
Comment (ehgmsdk20) [inline common/constants.js:141]: Thanks for adding the comments!

**Review by ehgmsdk20 — APPROVED** [review]
LGTM again, thanks for the nice job!

File: common/network-util.js (hunk 1 above)
Comment (minsulee2) [reply common/network-util.js:14]: Ohhh, I see. Thanks for the answer !

File: common/network-util.js [hunk 4]
```diff
@@ -108,6 +108,17 @@ function getIpAddress(internal = false) {
...
+    } else if (NodeConfigs.HOSTING_ENV === HostingEnvs.AWS) {
+      return axios.get(internal ? AWS_INTERNAL_IP_URL : AWS_EXTERNAL_IP_URL, {
+        timeout: 3000
+      })
+      .then((res) => {
+        return res.data;
+      })
+      .catch((err) => {
+        CommonUtil.finishWithStackTrace(
+            logger, `Failed to get ip address: ${JSON.stringify(err, null, 2)}`);
```
Comment (minsulee2) [inline common/network-util.js:120]: Cuz the logger is stackdriver, it won't work all envs but GCP I guess. Seems necessary to update it sometime soon.

File: common/constants.js [hunk 5]
```diff
@@ -821,6 +833,7 @@ module.exports = {
   TimerFlagEnabledBandageMap,
   BlockchainParams,
   NodeConfigs,
+  HostingEnvs,
```
Comment (minsulee2) [inline common/constants.js:836]: Love it !

**Review by minsulee2 — APPROVED** [review]
LGTM with minor comment :)

File: common/network-util.js (hunk 4 above)
Comment (platfowner) [reply common/network-util.js:120]: I think the bunyan/winston loggers are added only in gcp environment. Let me know if it's not correct.

https://github.com/ainblockchain/ain-blockchain/blob/master/logger/bunyan-util.js#L63
https://github.com/ainblockchain/ain-blockchain/blob/master/logger/winston-util.js#L105

**Comment by platfowner** [comment]
Thanks for the review. Now merging..

**Pull request merged**